File: QuanLyPhongKham.Server/src/QuanLyPhongKham.DTOs/Requests/DonThuocRequest.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QuanLyPhongKham.DTOs.Requests
{
    public class DonThuocRequest
    {
        public Guid? MaDT { get; set; }

        [Required(ErrorMessage = "Mã phiếu khám là bắt buộc.")]
        public Guid MaPK { get; set; }

        [Required(ErrorMessage = "Danh sách thuốc không được rỗng.")]
        [MinLength(1, ErrorMessage = "Phải có ít nhất 1 thuốc.")]
        public List<ChiTietDonThuocRequest> ChiTietDonThuocs { get; set; } = [];
    }
}

File: QuanLyPhongKham.Client/src/interfaces/requests/DonThuocRequest.js
/**
 * @typedef {Object} DonThuocRequest
 * @property {string} maDT
 * @property {string} maPK
 * @property {Array} chiTietDonThuocs
 */

export const createDonThuocRequest = (
  maDT = "",
  maPK = "",
  chiTietDonThuocs = [],
) => {
  const request = {
    maPK,
    chiTietDonThuocs,
  };
  if (maDT) request.maDT = maDT;
  return request;
};
